### PR TITLE
feat(content-explorer): Allow selecting empty selection in ContentExplorer

### PR DESCRIPTION
**Pull request by apieceofbart**
In Admin console we would like to support a case for choosing empty selection in the ContentExplorer. 
The reason for this is we keep the selection in local state and pass it to the ContentExplorer as initial selection. Therefore it is possible user would like to unselect everything and save this. Currently "Save" (choose) button is disabled if the selection is empty. 
We propose to add a boolean prop `allowChoosingEmptySelection` that when true would allow for selecting even when selection is empty.

### CODE REVIEW

File: src/features/content-explorer/content-explorer-modal-container/ContentExplorerModalContainer.js [hunk 1]
```diff
@@ -13,6 +13,8 @@ class ContentExplorerModalContainer extends Component {
          * Each column has to be a Column element
          */
         additionalColumns: PropTypes.arrayOf(PropTypes.element),
+        /**  Should user be able to choose empty selection in MULTI_SELECT mode, defaults to false */
+        allowChoosingEmptySelection: PropTypes.bool,
```
Comment (greg-in-a-box) [inline src/features/content-explorer/content-explorer-modal-container/ContentExplorerModalContainer.js:17]: Can we change the name of this prop using a similar convention to other props (e.g. `isSelectAllAllowed`) in this component? I think `isNoSelectionAllowed` might be more appropriate

File: src/features/content-explorer/content-explorer/ContentExplorerActionButtons.js [hunk 2]
```diff
@@ -46,9 +47,13 @@ const ContentExplorerActionButtons = ({
...
 
-        if (onChooseClick && chosenItems.length > 0) {
+        if (onChooseClick && (chosenItems.length > 0 || allowChoosingEmptySelection)) {
             onChooseClick(chosenItems);
         }
+
+        // if (onChooseClick && chosenItems.length > 0) {
+        //     onChooseClick(chosenItems);
+        // }
     };
```
Comment (greg-in-a-box) [inline src/features/content-explorer/content-explorer/ContentExplorerActionButtons.js:58]: I think its safe to say that we can delete these comments

File: src/features/content-explorer/content-explorer/__tests__/ContentExplorer.test.js [hunk 3]
```diff
@@ -240,6 +240,22 @@ describe('features/content-explorer/content-explorer/ContentExplorer', () => {
                 expect(wrapper.find('ContentExplorerActionButtons').prop('areButtonsDisabled')).toBe(true);
             });
         });
+
+        test('should render enabled action buttons in MULTI_SELECT mode if selection is empty but allowChoosingEmptySelection is true', () => {
```
Comment (greg-in-a-box) [inline src/features/content-explorer/content-explorer/__tests__/ContentExplorer.test.js:244]:
```suggestion
        test('should render with action buttons enabled in MULTI_SELECT mode if there is no selection made when isNoSelectionAllowed is true', () => {
```

File: src/features/content-explorer/content-explorer-modal-container/ContentExplorerModalContainer.js [hunk 4]
```diff
@@ -13,6 +13,8 @@ class ContentExplorerModalContainer extends Component {
          * Each column has to be a Column element
          */
         additionalColumns: PropTypes.arrayOf(PropTypes.element),
+        /**  Should user be able to choose empty selection in MULTI_SELECT mode, defaults to false */
```
Comment (greg-in-a-box) [inline src/features/content-explorer/content-explorer-modal-container/ContentExplorerModalContainer.js:16]:
```suggestion
        /**  Allow users to choose no selections in MULTI_SELECT mode, defaults to false */
```

**Review by greg-in-a-box — CHANGES_REQUESTED** [review]
left some feedback

File: src/features/content-explorer/content-explorer/ContentExplorerActionButtons.js (hunk 2 above)
Comment (apieceofbart) [reply src/features/content-explorer/content-explorer/ContentExplorerActionButtons.js:58]: haha yes, thank you

File: src/features/content-explorer/content-explorer-modal-container/ContentExplorerModalContainer.js (hunk 1 above)
Comment (apieceofbart) [reply src/features/content-explorer/content-explorer-modal-container/ContentExplorerModalContainer.js:17]: I don't mind `isSelectNoneAllowed` but I think the difference between this and `isSelectAllAllowed` is that there's also **choose** action involved - not only selecting.

**Review by greg-in-a-box — APPROVED** [review]

**Review by tjuanitas — APPROVED** [review]

**Pull request merged**